ethereumetl/service/eth_contract_service.py

```diff
@@ -69,6 +69,14 @@ def is_erc721_contract(self, function_sighashes):
                c.implements_any_of('transfer(address,uint256)', 'transferFrom(address,address,uint256)') and \
                c.implements('approve(address,uint256)')
 
+    # https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1155.md
+    # https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol
+    def is_erc1155_contract(self, function_sighashes):
+        c = ContractWrapper(function_sighashes)
+        return c.implements('balanceOf(address,uint256)') and \
+               c.implements('safeTransferFrom(address,address,uint256,uint256,bytes)') and \
+               c.implements('safeBatchTransferFrom(address,address,uint256[],uint256[],bytes')
```

Comment:
it should be ``c.implements('safeBatchTransferFrom(address,address,uint256[],uint256[],bytes)')`` 
(Add a `(` after the ``bytes``)

Reply:
I think the part which is relevant to 'token_transfer' also needs to support ERC 1155.